setup/source_pyrobosim.bash

```diff
@@ -14,49 +14,50 @@
 #  So you can then run this from your Terminal:
 #    pyrobosim
 #
-# For ROS workflows, enter the ROS distro (humble/iron) as an argument:
-#
-#   pyrobosim_ros() {
-#     source /path/to/pyrobosim/setup/source_pyrobosim.bash humble
-#   }
-#
-#   So you can then call run from your Terminal:
-#     pyrobosim_ros
 
-# User variables -- change this to meet your needs
-export VIRTUALENV_FOLDER=~/python-virtualenvs/pyrobosim
-export PYROBOSIM_WS=~/pyrobosim_ws
+# Set up the environment
+ENV_FILE="pyrobosim.env"
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+pushd "${SCRIPT_DIR}/.." > /dev/null
+if [ ! -f "${ENV_FILE}" ]
+then
+    popd > /dev/null
+    echo "Did not find a file named ${ENV_FILE} in the root pyrobosim folder."
+    echo "Please rerun the setup_pyrobosim.bash script."
+    return 1
+fi
+unset PYROBOSIM_VENV PYROBOSIM_ROS_WORKSPACE PYROBOSIM_ROS_DISTRO
+source "${ENV_FILE}"
+popd > /dev/null
 
 if [ -n "${VIRTUAL_ENV}" ]
 then
     deactivate
 fi
 
+# Activate the Python virtual environment
+echo "Activated virtual environment at ${PYROBOSIM_VENV}."
+source ${PYROBOSIM_VENV}/bin/activate
+
 # Parse ROS distro argument
-ROS_DISTRO=$1
-if [ "${ROS_DISTRO}" == "" ]
+if [ -z "${PYROBOSIM_ROS_WORKSPACE}" ]
 then
     echo "Setting up pyrobosim with no ROS distro."
 else
-    echo "Setting up pyrobosim with ROS ${ROS_DISTRO}."
-    source /opt/ros/${ROS_DISTRO}/setup.bash
-    pushd "${PYROBOSIM_WS}" > /dev/null || exit
-    if [ ! -d "build" ]
+    echo "Setting up pyrobosim with ROS ${PYROBOSIM_ROS_DISTRO}."
+    source /opt/ros/${PYROBOSIM_ROS_DISTRO}/setup.bash
+    pushd "${PYROBOSIM_ROS_WORKSPACE}" > /dev/null
```

Comment:
This directory either needs to be made earlier or you should set all these to fail on non-zero exit codes. If `pushd` fails it'll just build locally...

```
Activated virtual environment at /home/parallels/python-virtualenvs/pyrobosim.
Setting up pyrobosim with ROS humble.
bash: pushd: /home/parallels/pyrobosim_ws: No such file or directory
Building ROS workspace at /home/parallels/pyrobosim_ws...
Starting >>> pyrobosim_msgs
Finished <<< pyrobosim_msgs [6.93s]                     
Starting >>> pyrobosim_ros
Finished <<< pyrobosim_ros [1.16s]                  

Summary: 2 packages finished [8.16s]
Sourcing ROS workspace at /home/parallels/pyrobosim_ws.
bash: popd: directory stack empty
```

And then I just have build/install/log right in the repo

Reply:
Good call! I am now checking for the existence of this folder.